Applications/YOLO/jni/yolo_v2_loss.cpp

```diff
@@ -0,0 +1,432 @@
+  wt_idx[YoloV2LossParams::confidence_gt] = context.requestTensor(
+    confidence_gt_dim, "confidence_gt", nntrainer::Tensor::Initializer::NONE,
+    true, nntrainer::TensorLifespan::ITERATION_LIFESPAN);
+
+  nntrainer::TensorDim class_gt_dim(batch_size,
+                                    grid_height_number * grid_width_number,
+                                    max_object_number, class_number);
+  wt_idx[YoloV2LossParams::class_gt] = context.requestTensor(
+    class_gt_dim, "class_gt", nntrainer::Tensor::Initializer::NONE, true,
+    nntrainer::TensorLifespan::ITERATION_LIFESPAN);
+
+  nntrainer::TensorDim bbox_class_mask_dim(
+    batch_size, grid_height_number * grid_width_number, max_object_number, 1);
+  wt_idx[YoloV2LossParams::bbox_class_mask] =
+    context.requestTensor(bbox_class_mask_dim, "bbox_class_mask",
+                          nntrainer::Tensor::Initializer::NONE, false,
+                          nntrainer::TensorLifespan::ITERATION_LIFESPAN);
+
+  nntrainer::TensorDim iou_mask_dim(
+    batch_size, grid_height_number * grid_width_number, max_object_number, 1);
+  wt_idx[YoloV2LossParams::iou_mask] = context.requestTensor(
+    iou_mask_dim, "iou_mask", nntrainer::Tensor::Initializer::NONE, false,
+    nntrainer::TensorLifespan::ITERATION_LIFESPAN);
+}
+
+void YoloV2LossLayer::forwarding(nntrainer::RunLayerContext &context,
+                                 bool training) {
+  const unsigned int max_object_number =
+    std::get<props::MaxObjectNumber>(yolo_v2_loss_props).get();
+
+  nntrainer::Tensor &input = context.getInput(SINGLE_INOUT_IDX);
+  nntrainer::Tensor &label = context.getLabel(SINGLE_INOUT_IDX);
+
+  std::vector<nntrainer::Tensor> splited_input =
+    input.split({1, 1, 1, 1, 1, max_object_number}, 3);
+  nntrainer::Tensor bbox_x_pred_ = splited_input[0];
+  nntrainer::Tensor bbox_y_pred_ = splited_input[1];
+  nntrainer::Tensor bbox_w_pred_ = splited_input[2];
+  nntrainer::Tensor bbox_h_pred_ = splited_input[3];
+  nntrainer::Tensor confidence_pred_ = splited_input[4];
+  nntrainer::Tensor class_pred_ = splited_input[5];
+
+  nntrainer::Tensor &bbox_x_pred =
+    context.getTensor(wt_idx[YoloV2LossParams::bbox_x_pred]);
+  nntrainer::Tensor &bbox_y_pred =
+    context.getTensor(wt_idx[YoloV2LossParams::bbox_y_pred]);
+  nntrainer::Tensor &bbox_w_pred =
+    context.getTensor(wt_idx[YoloV2LossParams::bbox_w_pred]);
+  nntrainer::Tensor &bbox_h_pred =
+    context.getTensor(wt_idx[YoloV2LossParams::bbox_h_pred]);
+
+  nntrainer::Tensor &confidence_pred =
+    context.getTensor(wt_idx[YoloV2LossParams::confidence_pred]);
+  nntrainer::Tensor &class_pred =
+    context.getTensor(wt_idx[YoloV2LossParams::class_pred]);
+
+  nntrainer::Tensor &bbox_w_pred_anchor =
+    context.getTensor(wt_idx[YoloV2LossParams::bbox_w_pred_anchor]);
+  nntrainer::Tensor &bbox_h_pred_anchor =
+    context.getTensor(wt_idx[YoloV2LossParams::bbox_h_pred_anchor]);
+
+  bbox_x_pred.copyData(bbox_x_pred_);
```

Comment:
I wonder these copys are necessary. It might be better to consider removing them in later pr.